test/k4FWCoreTest/src/components/ExampleParticleIDConsumer.cpp

```diff
@@ -0,0 +1,154 @@
+      fatal() << fmt::format(
+                     "The PID algorithm name from metadata does not match the expected one from the properties: "
+                     "(expected {}, actual {})",
+                     algoName.value(), pidMeta.algoName)
+              << endmsg;
+      return false;
+    }
+
+    if (!std::ranges::equal(pidMeta.paramNames, paramNames)) {
+      fatal() << fmt::format(
+                     "The PID parameter names retrieved from metadata does not match the expected ones from the "
+                     "properties: (expected {}, actual {})",
+                     paramNames.value(), pidMeta.paramNames)
+              << endmsg;
+      return false;
+    }
+
+    return true;
+  }
+
+  void checkPIDForAlgo(const edm4hep::utils::PIDHandler& pidHandler, const edm4hep::ReconstructedParticle& reco,
+                       const edm4hep::utils::ParticleIDMeta& pidMeta, const int paramIndex) const {
+    auto maybePID = pidHandler.getPID(reco, pidMeta.algoType());
+    if (!maybePID) {
+      throw std::runtime_error(
+          fmt::format("Could net retrieve the {} PID object for reco particle {}", pidMeta.algoName, reco.id().index));
+    }
+    auto pid      = maybePID.value();
+    auto paramVal = pid.getParameters()[paramIndex];
+
+    // As set in the producer
+    if (paramVal != paramIndex * 0.5f) {
+      throw std::runtime_error(
+          fmt::format("Could not retrieve the correct parameter value for param {} (expected {}, actual {})",
+                      pidMeta.paramNames[paramIndex], paramIndex * 0.5f, paramVal));
+    }
+  }
+
+  StatusCode initialize() final {
+    m_pidMeta1 =
+        k4FWCore::getParameter<edm4hep::utils::ParticleIDMeta>(inputLocations("ParticleIDCollection1")[0]).value();
+
+    m_pidMeta2 =
+        k4FWCore::getParameter<edm4hep::utils::ParticleIDMeta>(inputLocations("ParticleIDCollection2")[0]).value();
+
+    if (!checkAlgoMetadata(m_pidMeta1, m_pidAlgoName1, m_pidParamNames1) ||
+        !checkAlgoMetadata(m_pidMeta2, m_pidAlgoName2, m_pidParamNames2)) {
+      return StatusCode::FAILURE;
+    }
+
+    m_paramIndex1 = edm4hep::utils::getParamIndex(m_pidMeta1, m_paramOfInterest1.value()).value_or(-1);
+    m_paramIndex2 = edm4hep::utils::getParamIndex(m_pidMeta2, m_paramOfInterest2.value()).value_or(-1);
+    if (m_paramIndex1 == -1 || m_paramIndex2 == -1) {
+      error() << fmt::format("Could not get a parameter index for {} (got {}) or {} (got {})",
+                             m_paramOfInterest1.value(), m_paramIndex1, m_paramOfInterest2.value(), m_paramIndex2)
+              << endmsg;
+    }
+
+    return StatusCode::SUCCESS;
+  }
+
+  void operator()(const edm4hep::ParticleIDCollection& pidColl1, const edm4hep::ParticleIDCollection& pidColl2,
+                  const edm4hep::ReconstructedParticleCollection& recos) const {
+    auto pidHandler = edm4hep::utils::PIDHandler::from(pidColl1, pidColl2);
+    pidHandler.addMetaInfos(m_pidMeta1, m_pidMeta2);
```

Comment:
This requires a new version of EDM4hep. Should I go for a version that works with 0.99? Otherwise we need to bump the EDM4hep version, I think. Even if this is only a test.

Reply:
Up to you. But actually 0.99.1 in EDM4hep has the problem that https://github.com/key4hep/EDM4hep/pull/379 is not merged in and this causes issues when trying to build on top of EDM4hep 0.99.1 since it won't find `nlohmann_json`.

Reply:
I have changed it such that CI still passes on the `release`s.